### PR TITLE
LBSD-1158: DELETE support on syndication webhook

### PR DESCRIPTION
**Pull request by mscam**
- Added DELETE support to syndication webhook
- Syndication webhook view refactoring

### CODE REVIEW

**Comment by mscam** [comment]
_producer_post_ has been changed to _post_. Unit tests for create, update and delete actions will be provided in a separate PR

**Comment by m0g** [comment]
LGTM :+1:

**Comment by m0g** [comment]
Actually I just tried it, and I'm getting this:

```
12:03:51 work.1 | API DELETE request to http://undefined.local:5000/api/syndication/webhook - response: 401 b'{"_error": {"code": 401, "message": "Authorization failed."}, "_status": "ERR"}' level=WARNING process=Worker-3
12:03:51 work.1 | API DELETE request to http://undefined.local:5000/api/syndication/webhook - response: 401 b'{"_error": {"code": 401, "message": "Authorization failed."}, "_status": "ERR"}' level=WARNING process=Worker-4
12:03:51 work.1 | Expecting value: line 1 column 1 (char 0) level=ERROR process=Worker-1
12:03:51 work.1 | Traceback (most recent call last):
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/celery/app/trace.py", line 240, in trace_task
12:03:51 work.1 |     R = retval = fun(*args, **kwargs)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/src/superdesk-core/superdesk/celery_app.py", line 113, in __call__
12:03:51 work.1 |     return super().__call__(*args, **kwargs)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/celery/app/trace.py", line 438, in __protected_call__
12:03:51 work.1 |     return self.run(*args, **kwargs)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/liveblog/syndication/tasks.py", line 24, in send_post_to_consumer
12:03:51 work.1 |     }, action)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/liveblog/syndication/consumer.py", line 91, in send_post
12:03:51 work.1 |     data=new_post)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/liveblog/syndication/consumer.py", line 77, in _send_api_request
12:03:51 work.1 |     timeout=timeout)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/liveblog/syndication/utils.py", line 88, in send_api_request
12:03:51 work.1 |     return response.json()
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/requests/models.py", line 763, in json
12:03:51 work.1 |     return json.loads(self.text, **kwargs)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/simplejson/__init__.py", line 516, in loads
12:03:51 work.1 |     return _default_decoder.decode(s)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/simplejson/decoder.py", line 370, in decode
12:03:51 work.1 |     obj, end = self.raw_decode(s)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/simplejson/decoder.py", line 400, in raw_decode
12:03:51 work.1 |     return self.scan_once(s, idx=_w(s, idx).end())
12:03:51 work.1 | simplejson.scanner.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
12:03:51 work.1 | Task liveblog.syndication.tasks.send_post_to_consumer[25afc55a-fa15-42e3-9e61-4b005341fcf9] raised unexpected: JSONDecodeError('Expecting value: line 1 column 1 (char 0)',) level=ERROR process=MainProcess
12:03:51 work.1 | Traceback (most recent call last):
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/celery/app/trace.py", line 240, in trace_task
12:03:51 work.1 |     R = retval = fun(*args, **kwargs)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/src/superdesk-core/superdesk/celery_app.py", line 113, in __call__
12:03:51 work.1 |     return super().__call__(*args, **kwargs)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/celery/app/trace.py", line 438, in __protected_call__
12:03:51 work.1 |     return self.run(*args, **kwargs)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/liveblog/syndication/tasks.py", line 24, in send_post_to_consumer
12:03:51 work.1 |     }, action)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/liveblog/syndication/consumer.py", line 91, in send_post
12:03:51 work.1 |     data=new_post)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/liveblog/syndication/consumer.py", line 77, in _send_api_request
12:03:51 work.1 |     timeout=timeout)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/liveblog/syndication/utils.py", line 88, in send_api_request
12:03:51 work.1 |     return response.json()
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/requests/models.py", line 763, in json
12:03:51 work.1 |     return json.loads(self.text, **kwargs)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/simplejson/__init__.py", line 516, in loads
12:03:51 work.1 |     return _default_decoder.decode(s)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/simplejson/decoder.py", line 370, in decode
12:03:51 work.1 |     obj, end = self.raw_decode(s)
12:03:51 work.1 |   File "/home/loic/code/liveblog/server/env/lib/python3.5/site-packages/simplejson/decoder.py", line 400, in raw_decode
12:03:51 work.1 |     return self.scan_once(s, idx=_w(s, idx).end())
12:03:51 work.1 | simplejson.scanner.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
12:03:52 rest.1 | INFO:werkzeug:192.168.114.52 - - [08/Dec/2016 12:03:52] "OPTIONS /api/activity?embedded=%7B%22user%22:1,%22item%22:1%7D&max_results=8&where=%7B%22read.5788bc55ea8e157ca2ab7629%22:%7B%22$exists%22:true%7D%7D HTTP/1.1" 200 -
12:03:52 rest.1 | INFO:werkzeug:192.168.114.52 - - [08/Dec/2016 12:03:52] "GET /api/activity?embedded=%7B%22user%22:1,%22item%22:1%7D&max_results=8&where=%7B%22read.5788bc55ea8e157ca2ab7629%22:%7B%22$exists%22:true%7D%7D HTTP/1.1" 200 -

```

Am I forgetting something?

**Comment by mscam** [comment]
Need to fix http status error check to prevent json decode

File: server/liveblog/syndication/syndication.py [hunk 1]
```diff
@@ -169,36 +171,53 @@ class SyndicationIn(Resource):
...
 
-        post_id = str(post['_id'])
-        publisher = post.get('publisher')
-        if not publisher:
-            # Update only posts with publisher set to None
             posts_service.update(post_id, new_post, post)
             return api_response({'post_id': post_id}, 200)
-        else:
-            return api_error('Post "{}" cannot be updated: publisher already exists.'.format(post_id), 409)
+    else:
```
Comment (nistormihai) [inline server/liveblog/syndication/syndication.py]: I think a check on method 
`if request.method in ('DELETE'):` 
it is clear and if additional methods are added no more refactor on this part.

**Comment by nistormihai** [comment]
LGTM

**Pull request merged**